### ISSUE TITLE
Send failure if to and/or cc had duplicate address

### ISSUE THREAD

**Issue by dsg18096** [k9mail/k-9#4781]
Please search to check for an existing issue (including closed issues, for which the fix may not have yet been released) before opening a new issue: https://github.com/k9mail/k-9/issues?q=is%3Aissue

### Expected behavior
Tell us what should happen

I reply all to an email in which the sender has cc'd him or herself. This results in the sender's email showing up twice in the header.  I expect the delicate to be ignored, as it is in every other email client I've used in the past 20 years or more.

### Actual behavior
Tell us what happens instead

Message gets stuck in outbox.  At least the error message explains it and I can edit it again to remove the duplicate and resend.
### Steps to reproduce
1. Reply all to an email on which the sender has cc'd herself
2. Note that sender's email is in the header twice, once in cc, once in to
3. Send, now that message is stuck in outbox with an error

### Environment
K-9 Mail version:
5.714
Android version:
10
Account type (IMAP, POP3, WebDAV/Exchange):
Imap
Please take some time to [retrieve logs](https://github.com/k9mail/k-9/wiki/LoggingErrors) and attach them here:

**Comment by cketti** [comment]
PR #4788 contains a change that will probably fix the send error. It won't change the headers to remove duplicates.

**Comment by dsg18096** [comment]
It happened again tonight. I hadn't noticed it in a long while and thought that the charset fix must have solved it.  The issue tonight was someone replying to a Google group message. The From was the group address so my reply all included the group address on the To and CC.  When I sent, it got stuck in the outbox with the duplicate recipient error message. I was able to edit it as a new message to remove the second copy of the address and it went but a blank message was also sent.

**Comment by life777eternal** [comment]
@dsg18096 I've noticed that when the phone screen is accidentally rotated to landscape and then back to portrait, email addresses in the To field are repeated once and in the CC & BCC fields it is repeated twice. In plain text, but they kind of disappear after touching another part of the app. Is that what you've been noticing? 

K-9 Version: 16.1

Android Version: 15

**Comment by dsg18096** [comment]
@life777eternal No, that is not what I noticed 5 years ago when I opened this :-)  As far as I can tell, the issue that I was concerned about has been fixed.  At least, I haven't had to edit a stalled message in outbox to remove duplicate email addresses in a very long time.
